unyt/tests/test_array_functions.py

```diff
@@ -1371,4 +1366,201 @@ def test_pad():
+    np.put(arr, [1, 2], [44, 55] * cm)
+    assert type(arr) is unyt_array
+    assert arr.units == cm
+
+
+def test_put_along_axis_mixed_units():
+    arr = np.arange(6).reshape(2, 3) * cm
+    with pytest.raises(UnitInconsistencyError):
+        np.put_along_axis(arr, np.array([[1, 2], [0, 1]]), [44, 55], 1)
+
+
+def test_put_along_axis():
+    arr = np.arange(6).reshape(2, 3) * cm
+    np.put_along_axis(arr, np.array([[1, 2], [0, 1]]), [44, 55] * cm, 1)
+    assert type(arr) is unyt_array
+    assert arr.units == cm
+
+
+def test_putmask_mixed_units():
+    arr = np.arange(6, dtype=np.int_).reshape(2, 3) * cm
+    with pytest.raises(UnitInconsistencyError):
+        np.putmask(arr, arr > 2 * cm, np.zeros_like(arr.d))
+
+
+def test_putmask():
+    arr = np.arange(6, dtype=np.int_).reshape(2, 3) * cm
+    np.putmask(arr, arr > 2 * cm, np.zeros_like(arr))
+
+    assert type(arr) is unyt_array
+    assert arr.units == cm
+
+
+def test_searchsorted_mixed_units():
+    with pytest.raises(UnitInconsistencyError):
+        np.searchsorted([1, 2, 3, 4, 5] * cm, 3)
+
+
+def test_searchsorted():
+    res = np.searchsorted([1, 2, 3, 4, 5] * cm, 3 * cm)
+    assert res == 2
+
+
+@pytest.mark.parametrize(
+    "choicelist_gen, default",
+    [
+        ([lambda x: x**3, lambda x: x**2], 34 * cm),  # invalid choicelist
+        ([lambda x: x, lambda x: x + 3 * x.units], 34),  # invalid default
+    ],
+)
+def test_select_mixed_units(choicelist_gen, default):
+    a = [1, 2, 3, 4, 5, 6] * cm
+    with pytest.raises(UnitInconsistencyError):
+        np.select(
+            [a > 3 * cm, a < 3 * cm], [f(a) for f in choicelist_gen], default=default
+        )
+
+
+def test_select():
+    a = [1, 2, 3, 4, 5, 6] * cm
+    res = np.select([a > 3 * cm, a < 3 * cm], [a, a + 3 * cm], default=34 * cm)
+    assert_array_equal_units(res, [4, 5, 34, 4, 5, 6] * cm)
+
+
+def test_setdiff1d_mixed_units():
+    a = [1, 2, 3] * cm
+    b = [0, 1, 2]
+    with pytest.raises(UnitInconsistencyError):
+        np.setdiff1d(a, b)
+
+
+def test_setdiff1d():
+    a = [1, 2, 3] * cm
+    b = [0, 1, 2] * cm
+    res = np.setdiff1d(a, b)
+    assert_array_equal_units(res, [3] * cm)
+
+
+def test_setxor1d_mixed_units():
+    a = [1, 2, 3] * cm
+    b = [0, 1, 2]
+    with pytest.raises(UnitInconsistencyError):
+        np.setxor1d(a, b)
+
+
+def test_setxor1d():
+    a = [1, 2, 3] * cm
+    b = [0, 1, 2] * cm
+    res = np.setxor1d(a, b)
+    assert_array_equal_units(res, [0, 3] * cm)
+
+
+def test_where_mixed_units():
+    x = [-1, 2, -3] * cm
+    y = [0, 0, 0]
+    with pytest.raises(UnitInconsistencyError):
+        np.where(x > y, x, y)
+
+
+def test_where():
+    x = [-1, 2, -3] * cm
+    y = [0, 0, 0] * cm
+    res = np.where(x > y, x, y)
+    assert type(res) is unyt_array
+    assert res.units == cm
+
+
+def test_clip_mixed_units():
+    a = [1, 2, 3, 4, 5, 6] * cm
+    with pytest.raises(UnitInconsistencyError):
+        np.clip(a, 3 * cm, 4)
```

Comment:
I agree this should error, but I kinda think `np.clip(a, 3, 4)` shouldn't error. The fact that it does now breaks existing unyt tests is another indicator that it shouldn't error, since it could just as easily break downstream user code.

Reply:
>  but I kinda think np.clip(a, 3, 4) shouldn't error

me too TBH, but in that case it seems to me that making `np.clip(a, 3 * cm, 4)` error will add more complexity than it is worth.

Reply:
nevermind it's actually quite simple